### DISCUSSION TITLE
How to download Chrome Dev tools Performance Monitoring or JS Profiler report/log json while running Cypress

### DISCUSSION THREAD

**Discussion by prasanthkarukkuvel** [cypress-io/cypress#25412] [Questions and Help] [answered]
Is there is an option to download the performance report from Cypress or using any other plugins? I am looking for very similar implementation with puppeteer where we can start and stop the tracing programmatically.

**Comment by portyu9** [comment, marked as answer]
Hi @prasanthkarukkuvel 

Cypress still does not expose a first-class Puppeteer-style tracing API that can start/stop Chrome Performance tracing and return the DevTools trace JSON.

You can send individual Chromium CDP commands through the low-level `Cypress.automation('remote:debugger:protocol', ...)` hook, but a complete `Tracing.start` / `Tracing.end` workflow also requires receiving CDP events such as `Tracing.tracingComplete` and handling the resulting trace data/stream.

Cypress does not currently expose that CDP event-listening interface. A later Cypress feature request for exactly this use case was closed as a duplicate of the long-standing request to expose the full Chrome DevTools Protocol API.

So the practical choices are:

- If you need an actual Chrome DevTools trace / CPU-profile JSON, use Puppeteer, Playwright, or a separate CDP profiling step for that performance workflow.
- If you only need application-level performance measurements in Cypress, collect data from `window.performance` / `PerformanceObserver` and persist it with `cy.writeFile()` or `cy.task()`.

I would not build trace-export tooling around the undocumented `Cypress.automation('remote:debugger:protocol')` hook because sending commands alone is not sufficient for this use case.

Please mark this answer as accepted if it helped, thank you.
